Command/PasswordResetHandler.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace SumoCoders\FrameworkMultiUserBundle\Command;
+
+use SumoCoders\FrameworkMultiUserBundle\Exception\InvalidPasswordConfirmationException;
+use SumoCoders\FrameworkMultiUserBundle\User\UserRepositoryCollection;
+
+class PasswordResetHandler
+{
+    /**
+     * @var UserRepositoryCollection
+     */
+    private $userRepositoryCollection;
+
+    /**
+     * PasswordResetHandler constructor.
+     *
+     * @param UserRepositoryCollection $userRepositoryCollection
+     */
+    public function __construct(UserRepositoryCollection $userRepositoryCollection)
+    {
+        $this->userRepositoryCollection = $userRepositoryCollection;
+    }
+
+    /**
+     * @param ResetPassword $command
+     *
+     * @throws InvalidPasswordConfirmationException
+     */
+    public function handle(ResetPassword $command)
+    {
+        if (!$command->passwordConfirmationIsValid()) {
+            throw new InvalidPasswordConfirmationException('The password confirmation isn\'t valid');
+        }
+        
+        $user = $command->getUser();
+        $updatedUser = clone $user;
```

Comment:
wat is het reden van het clonen hier precies? Je wilt toch de huidige user aanpassen, niet een extra instantie maken?